examples/exex/discv5/src/network/mod.rs

```diff
@@ -0,0 +1,110 @@
+use discv5::{enr::secp256k1::rand, ListenConfig};
+use futures_util::StreamExt;
+use reth::{
+    network::{
+        config::SecretKey, NetworkConfig, NetworkEvent, NetworkEvents, NetworkHandle,
+        NetworkManager,
+    },
+    primitives::{ChainSpec, Genesis},
+    providers::test_utils::NoopProvider,
+};
+use reth_tracing::tracing::info;
+use std::{
+    future::Future,
+    net::{SocketAddrV4, SocketAddrV6},
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+pub(crate) mod cli_ext;
+
+pub(crate) struct Discv5Network {
+    handle: NetworkHandle,
+}
+
+impl Discv5Network {
+    pub async fn new(ipv4_addr: SocketAddrV4, ipv6_addr: SocketAddrV6) -> Self {
+        let secret_key = SecretKey::new(&mut rand::thread_rng());
+        let discv5_config = discv5::ConfigBuilder::new(ListenConfig::from_two_sockets(
+            Some(ipv4_addr),
+            Some(ipv6_addr),
+        ))
+        .build();
+
+        let net_cfg = NetworkConfig::builder(secret_key)
```

Comment:
I think we don't want a full network here and only spawn discv5.

this exex will be the foundation for new features that then actually make use of this discovery network